frame/support/procedural/src/construct_runtime/mod.rs

```diff
@@ -389,8 +389,9 @@ fn decl_all_pallets<'a>(
 				(attr, names)
 			} else {
 				let test_cfg = features.remove("test").then_some(quote!(test)).into_iter();
+				let disabled_features = all_features.difference(&features);
 				let features = features.iter();
-				let attr = quote!(#[cfg(all( #(#test_cfg),* #(feature = #features),* ))]);
+				let attr = quote!(#[cfg(all( #(#test_cfg),* #(feature = #features,)* #(not(feature = #disabled_features)),* ))]);
```

Comment:
```suggestion
				let attr = quote!(#[cfg(all( #(#test_cfg,)* #(feature = #features,)* #(not(feature = #disabled_features)),* ))]);
```

Should have been wrong before, because we need here a comma when `test_cfg` is set.